lib/matplotlib/dates.py

```diff
@@ -207,16 +207,19 @@ def _get_rc_timezone():
 
 def _to_ordinalf(dt):
     """
-    Convert :mod:`datetime` or :mod:`date` to the Gregorian date as UTC float
-    days, preserving hours, minutes, seconds and microseconds.  Return value
-    is a :func:`float`.
+    Convert :mod:`datetime`, :mod:`date` or :mod:`timedelta` to the Gregorian
+    date as UTC float days, preserving hours, minutes, seconds and
+    microseconds. Return value is a :func:`float`.
     """
     # Convert to UTC
     tzi = getattr(dt, 'tzinfo', None)
     if tzi is not None:
         dt = dt.astimezone(UTC)
         tzi = UTC
 
+    if isinstance(dt, datetime.timedelta):
```

Comment:
@WeatherGod I expressed my reservations a bit more in [this comment](https://github.com/matplotlib/matplotlib/issues/8869#issuecomment-323498027). `timedelta` is a weird animal in that you normally don't have separate units for `x` and `delta-x`, but in datetimes you do.

One problem with this as a solution to #4916 is that a corollary to #4916 is that it seems to indicates that `datetime` *could* be used as a width in `rect`, which is just as wrong as `timedelta` failing to work (and this does *not* fix that problem).

I don't know enough about the way the width processing and unit framework to know what the right seam is, but I would think that the right place to do the type check would be earlier in the pipeline, when you still know whether this is supposed to be units of `x` or `delta-x`.